website/server/models/group.js

```diff
@@ -724,6 +724,10 @@ schema.methods.leave = async function leaveGroup (user, keep = 'keep-all') {
     pusher.trigger(`presence-group-${group._id}`, 'user-left', {
       userId: user._id,
     });
+
+    delete group.quest.members[user._id];
+    group.markModified('quest.members');
+    promises.push(group.save());
```

Comment:
Beat me to it, but yes to what you said. Only use `$unset` instead of `$set`. We don't need to overwrite the whole `quest.members` object, just the one member.